### ISSUE TITLE
flutter symbolize does not deobfuscate

### ISSUE THREAD

**Issue by easeccy** [flutter/flutter#101910]
I have this StackTrace on Firebase Crashlytics.
<img width="799" alt="image" src="https://user-images.githubusercontent.com/26852018/163404867-628f4061-b03e-465b-b1d1-196838d4bd07.png">
I deleted the . prefix and (.java) parts at the end of each line.
```
#00 abs 0 virt 000000000066246b _kDartIsolateSnapshotInstructions+0x4856fb
#01 abs 0 virt 0000000000663523 _kDartIsolateSnapshotInstructions+0x4867b3
#02 abs 0 virt 0000000000312f17 _kDartIsolateSnapshotInstructions+0x1361a7
#03 abs 0 virt 00000000005f75d7 _kDartIsolateSnapshotInstructions+0x41a867
```
When I run `flutter symbolize` it gives the same obfuscated output.
![image](https://user-images.githubusercontent.com/26852018/163405020-64952fe6-9144-4258-bf6b-2164e3b258f5.png)

Is this expected?

**Comment by darshankawar** [comment]
@easeccy 
Are you using `proguard` in your project ?

**Comment by easeccy** [comment]
No. Does it effect the Dart obfuscation of Dart stacktrace?

**Comment by darshankawar** [comment]
No exactly. There's an open issue wherein having proguard does not deobfuscate stacktrace.
I have a feeling that `flutter symbolize` would deobfuscate fatal exceptions and probably not deobfuscating non-fatal exception, which is your case.

I am keeping it open and labeling for further insights from the team.

/cc @christopherfujino

**Comment by cuzyoucant** [comment]
I had the same issue of symbolize not deobfuscating my stacktrace. In my case i obtained it via the Catcher plugin.

The reason is that the internat tool uses the Regex
'final _traceLineRE = RegExp(
    r'    #(\d+) abs (?<absolute>[\da-f]+)(?: virt (?<virtual>[\da-f]+))? '
    r'(?<rest>.*)$');'
to search the obfuscated stacktrace.
As you can see it expected 4 spaces infront of a line. After manually adding those 4 spaces before the # it worked as expected.

In my case Catcher returned the obfuscated stacktrace with only 1 space.

Might be a good idea to lighten the requirements of the regex. I dont see a need why the stacktrace would need to absolutely have 4 spaces infront.

Hope that helps!

**Comment by pawlowskim** [comment]
> I had the same issue of symbolize not deobfuscating my stacktrace. In my case i obtained it via the Catcher plugin.
> 
> The reason is that the internal tool uses the Regex (file: native_stack_traces-0.4.6/lib/src/convert.dart:72)
> 
> ```
> final _traceLineRE = RegExp(
>     r'    #(\d+) abs (?<absolute>[\da-f]+)(?: virt (?<virtual>[\da-f]+))? '
>     r'(?<rest>.*)$');
> ```
> 
> to search the obfuscated stacktrace. As you can see it expected 4 spaces infront of a line. After manually adding those 4 spaces before the # it worked as expected.
> 
> In my case Catcher returned the obfuscated stacktrace with only 1 space.
> 
> Might be a good idea to lighten the requirements of the regex. I dont see a need why the stacktrace would need to absolutely have 4 spaces infront.
> 
> Hope that helps!

Same for crashes caught by crashlytics for Android. iOS works just fine.
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.3.7, on macOS 12.6 21G115 darwin-x64, locale en-GB)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
[✓] Xcode - develop for iOS and macOS (Xcode 14.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.2)
[✓] IntelliJ IDEA Ultimate Edition (version 2022.1.4)
[✓] Connected device (3 available)
[✓] HTTP Host Availability

• No issues found!

firebase_crashlytics: 3.0.4

gradle crashlytics:
classpath 'com.google.firebase:firebase-crashlytics-gradle:2.5.2'
```

Adding 4 spaces workarounds the issue.

**Comment by sperochon** [comment]
It seems this bug have been corrected in native_stack_traces 0.5.5
https://pub.dev/packages/native_stack_traces/changelog
https://github.com/dart-lang/sdk/commit/73c639a4baef647c97db7f0ef62f6b1a9ca83631

which seems to be used only from Flutter 3.10+

I've tried `flutter symbolize` with Flutter 3.10 and I confirm it is now working fine (= de-obfuscate the stack trace)

**Comment by darshankawar** [comment]
Thanks @sperochon for the update and insights. @easeccy Check above comment and see if it helps to resolve the issue you reported.
